app/src/main/java/mozilla/lockbox/store/AccountStore.kt

```diff
@@ -194,16 +203,56 @@ open class AccountStore(
             .subscribe(profileSubject::onNext, this::pushError)
             .addTo(compositeDisposable)
 
+        val token = securePreferences.getString(Constant.Key.accessToken)?.let {
```

Comment:
I would instead introduce a helper class that handles token caching/rotation, and only touch a single line in the `AccountStore`. You will be able to trivially test that class's behaviour then, as well, and keep this logic simple.

Reply:
This is an excellent suggestion.